src/journal-remote/journal-upload-journal.c

```diff
@@ -284,14 +282,14 @@ static size_t journal_input_callback(void *buf, size_t size, size_t nmemb, void
                         u->entry_state = ENTRY_CURSOR;
                 }
 
-                w = write_entry((char*)buf + filled, size * nmemb - filled, u);
+                w = write_entry((char*)buf + offset + filled, size - filled, u);
                 if (w < 0)
-                        return CURL_READFUNC_ABORT;
+                        return w;
                 filled += w;
 
                 if (filled == 0) {
                         log_error("Buffer space is too small to write entry.");
-                        return CURL_READFUNC_ABORT;
+                        return -1;
```

Comment:
in our codebase it's not OK to make up literal error codes such as "-1". Always use propagate proper, negative errno-style error codes